### PR TITLE
Add model selector to `AI Text` block

### PR DESCRIPTION
**Pull request by luisbettencourt**
## 🌟 What is the purpose of this PR?

This PR adds the selector to the AI text block's header that allows picking the desired language model.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203358502199087/1203701786066059/f) _(internal)_

## 📹 Demo

image

### CODE REVIEW

File: blocks/ai-text/src/app/generate-text/model-selector.tsx [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+const MODELS_BY_GROUP: { [key in ModelGroupName]: ModelGroup } = {
+  [ModelGroupName.OpenAi]: {
+    name: ModelGroupName.OpenAi,
+    icon: <AbstractAiIcon sx={{ fontSize: "inherit" }} />,
+    models: [
+      {
+        id: "gpt-3.5-turbo",
+        name: "GPT-3.5 Turbo",
+        description:
+          "The best current language model for many use cases. Newer, faster and 10x cheaper than Davinci.",
```
Comment (nonparibus) [inline blocks/ai-text/src/app/generate-text/model-selector.tsx:40]:
```suggestion
          "The best current language model for many use cases; newer, faster and 10x cheaper than Davinci",
```

File: blocks/ai-text/src/app/generate-text/model-selector.tsx [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+const MODELS_BY_GROUP: { [key in ModelGroupName]: ModelGroup } = {
+  [ModelGroupName.OpenAi]: {
+    name: ModelGroupName.OpenAi,
+    icon: <AbstractAiIcon sx={{ fontSize: "inherit" }} />,
+    models: [
+      {
+        id: "gpt-3.5-turbo",
+        name: "GPT-3.5 Turbo",
+        description:
+          "The best current language model for many use cases; newer, faster and 10x cheaper than Davinci.",
```
Comment (nonparibus) [inline blocks/ai-text/src/app/generate-text/model-selector.tsx:40]:
```suggestion
          "The best model for many use cases; faster and 10x cheaper than Davinci",
```

File: blocks/ai-text/src/app/generate-text/model-selector.tsx (hunk 1 above)
Comment (nonparibus) [reply blocks/ai-text/src/app/generate-text/model-selector.tsx]: The `.` at the end was the other change here that got missed... however I have managed to shorten this a bit more. See https://github.com/hashintel/hash/pull/2176/files#r1129563430

**Comment by nonparibus** [comment]
Just spotted we're using the wrong `caret-down` icon in the link that expands the dropdown menu.

File: blocks/ai-text/src/app/generate-text/model-selector.tsx [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+} from "@mui/material";
+import { ReactNode, useMemo, useState } from "react";
+
+import { AbstractAiIcon } from "../../icons/abstract-ai";
+import { CheckIcon } from "../../icons/check";
+
+export const DEFAULT_MODEL_ID = "gpt-3.5-turbo";
+
+enum ModelGroupName {
+  OpenAi = "OpenAi",
```
Comment (nonparibus) [inline blocks/ai-text/src/app/generate-text/model-selector.tsx:17]:
```suggestion
  OpenAi = "OpenAI",
```

File: blocks/ai-text/src/dev.tsx [hunk 4]
```diff
@@ -22,8 +38,14 @@ const DevApp = () => {
...
       blockInfo={packageJSON.blockprotocol}
+      initialData={{
+        initialEntities: [
+          testEntity,
+          // ...imageEntities
+        ],
+      }}
       // @todo add dot-env support
-      blockProtocolApiKey={undefined} // Set this to an API key when testing
+      blockProtocolApiKey="b10ck5.43430b0bbd608928f03b8fe5aff452c8.fe03fe6b-e8ed-4725-8eae-adcb890af804" // Set this to an API key when testing
```
Comment (nonparibus) [inline blocks/ai-text/src/dev.tsx:48]: We probably want to strip this and regen since it's now been made public? @CiaranMn @luisbettencourt

Comment (CiaranMn) [reply blocks/ai-text/src/dev.tsx:48]: I have revoked it

Comment (nonparibus) [inline blocks/ai-text/src/dev.tsx:48]:
```suggestion
      blockProtocolApiKey={undefined} // Set this to an API key when testing
```

File: blocks/ai-text/src/app/generate-text/model-selector.tsx [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+                }}
+              >
+                {groupName}
+              </Typography>
+
+              <Stack>
+                {models.map(({ id, name, description }) => {
+                  const active = id === model;
+                  return (
+                    <MenuItem
```
Comment (CiaranMn) [inline blocks/ai-text/src/app/generate-text/model-selector.tsx:225]: This doesn't seem to be being picked up by the `Select` as an option – there are these warnings in the console, even if this is given a `value`. I'm guessing that's why there is a custom `onClick` here, and the `onChange` on the `Select` is never triggered.

There are also no keyboard controls available for the `Select`, probably related. Maybe we can use [grouping](https://mui.com/material-ui/react-select/#grouping) but customise how they're displayed?

```
SelectInput.js?6fd7:406 MUI: 
You have provided an out-of-range value `gpt-3.5-turbo` for the select component.
Consider providing a value that matches one of the available options or ''.
The available values are "".
```

**Review by CiaranMn — APPROVED** [review]

File: blocks/ai-text/src/dev.tsx (hunk 4 above)
Comment (luisbettencourt) [reply blocks/ai-text/src/dev.tsx]: Ah this file was never meant to be comitted, my bad!

**Review by CiaranMn — APPROVED** [review]

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**